core/src/main/java/com/graphhopper/GraphHopper.java

```
@@ -297,18 +297,29 @@ private GraphHopper setUnsafeMemory()
      * Enables the use of contraction hierarchies to reduce query times. Enabled by default.
      * <p>
      *
-     * @param weighting can be "fastest", "shortest" or your own weight-calculation type.
+     * @param weightingList A list containing multiple weightings like:
+     *                     "fastest", "shortest" or your own weight-calculation type.
      */
-    public GraphHopper setCHWeighting( String weighting )
+    public GraphHopper setCHWeighting( List<String> weightingList )
     {
         ensureNotLoaded();
-        chWeightingStr = weighting.toLowerCase();
+
+        if(weightingList.isEmpty())
```

we should throw an exception here saying at least one weighting is required.